themes/default/content/docs/guides/automation-api/getting-started-automation-api.md

```diff
@@ -0,0 +1,466 @@
+                            {
+                                Identifiers = new List<string> { "*" },
+                                Type = "AWS",
+                            },
+                        },
+                        Actions = new List<string> { "s3:GetObject" },
+                        Resources = new List<string> { $"{bucketArn}/*" },
+                    },
+                },
+            }));
+    });
+
+    // Set the access policy for the bucket so all objects are readable.
+    new Pulumi.Aws.S3.BucketPolicy(
+        "bucket-policy",
+        new Pulumi.Aws.S3.BucketPolicyArgs
+        {
+            Bucket = siteBucket.BucketName,
+            Policy = bucketPolicyDocument.Apply(x => x.Json),
+        });
+
+    // Export the website url.
+    return new Dictionary<string, object?>
+    {
+        ["website_url"] = siteBucket.WebsiteEndpoint,
+    };
+});
+```
+
+{{% /choosable %}}
+
+## Associate with a stack
+
+As with executing Pulumi programs through the CLI, you need to associate your Pulumi program with a `Stack`. Automation API provides methods to create or select stacks.
+
+Here's a convenient method to select an existing `Stack` or create one if none exists:
+
+{{< chooser language "javascript,typescript,python,go,csharp" >}}
+
+{{% choosable language "javascript,typescript" %}}
+
+```typescript
+const args: InlineProgramArgs = {
+    stackName: "dev",
+    projectName: "inlineNode",
+    program: pulumiProgram
+};
+
+const stack = await LocalWorkspace.createOrSelectStack(args);
+```
+
+{{% /choosable %}}
+
+{{% choosable language python %}}
+
+```python
+project_name = "inline_s3_project"
+stack_name = "dev"
+
+stack = auto.create_or_select_stack(stack_name=stack_name,
+                                    project_name=project_name,
+                                    program=pulumi_program)
+```
+
+{{% /choosable %}}
+
+{{% choosable language go %}}
+
+```go
+projectName := "inlineS3Project"
+stackName := "dev"
+s, err := auto.UpsertStackInlineSource(ctx, stackName, projectName, deployFunc)
+```
+
+{{% /choosable %}}
+
+{{% choosable language "csharp,fsharp,visualbasic" %}}
+
+```csharp
+var projectName = "inline_s3_project";
+var stackName = "dev";
+
+var stackArgs = new InlineProgramArgs(projectName, stackName, program);
+var stack = await LocalWorkspace.CreateOrSelectStackAsync(stackArgs);
+```
+
+{{% /choosable %}}
+{{< /chooser >}}
+
+A `Stack` operates within the context of a `Workspace`. A `Workspace` is the execution context containing a single Pulumi project, a program, and multiple stacks. Workspaces are used to manage the execution environment, providing various utilities such as plugin installation, environment configuration (`$PULUMI_HOME`), and creation, deletion, and listing of stacks. Because you are deploying AWS resources in this tutorial, you must install the AWS provider plugin within your `Workspace` so that your Pulumi program will have it available during execution.
```

Comment:
```suggestion
A stack operates within the context of a _workspace_. A workspace is the execution context containing a single Pulumi project, a program, and multiple stacks. Workspaces are used to manage the execution environment, providing various utilities such as plugin installation, environment configuration (`$PULUMI_HOME`), and creation, deletion, and listing of stacks. Because you are deploying AWS resources in this tutorial, you must install the AWS provider plugin within your workspace so that your Pulumi program will have it available during execution.
```

Reply:
The reason these words are capitalized and in "code marks" is because we are talking about the class `Stack` and `Workspace` specifically, rather than the more general concepts of "stack" and "workspace". Curious if that changes your feedback at all.